aten/src/ATen/native/sparse/mps/kernels/Mul.metal

```diff
@@ -0,0 +1,198 @@
+#include <metal_stdlib>
+#include <c10/metal/indexing.h>
+using namespace metal;
+
+
+template <typename T>
+kernel void dense_sparse_mul_kernel(
+    device const T* dense         [[buffer(0)]],
+    device const T* values        [[buffer(1)]],
+    device T* out_values          [[buffer(2)]],
+    device const long* indices    [[buffer(3)]],
+    device const long* sizes      [[buffer(4)]],
+    constant uint& nnz            [[buffer(5)]],
+    constant uint& nDimI          [[buffer(6)]],
+    constant uint& view_cols      [[buffer(7)]],
```

Comment:
We've talked in the high is the overhead of passing one scalar as argument buffer. Please merge those in followup PR
```suggestion
    constant uint3& nnz_dim_viewcols            [[buffer(5)]],
```